sound/soc/intel/boards/skl_hda_dsp_common.c

```diff
@@ -132,6 +132,11 @@ int skl_hda_hdmi_jack_init(struct snd_soc_card *card)
 		if (err)
 			return err;
 
+		err = snd_jack_add_new_kctl(pcm->hdmi_jack.jack,
+					    jack_name, SND_JACK_AVOUT);
+		if (err)
+			dev_warn(component->dev, "failed creating Jack kctl\n");
+
```

Comment:
can you clarify for me how this is released? there is no matching 'release' for the 'new'.

Also should we add those controls once the hdac_hdmi_jack_init() succceds below, otherwise you may expose controls that are not functional.

Reply:
> can you clarify for me how this is released? there is no matching 'release' for the 'new'.
> 
> Also should we add those controls once the hdac_hdmi_jack_init() succceds below, otherwise you may expose controls that are not functional.

@plbossart ALSA jack has a mechanism that it will release them automatically. Jack is seemed as a snd_device. So when driver is removed, the snd_deivce's dev_free() will be called by alsa core. In Jack's dev_free(), it will free all the resource. So you can seem there is no free kctl code in our audio driver. You can refer jack.c: snd_jack_dev_free()

Reply:
> Also should we add those controls once the hdac_hdmi_jack_init() succceds below, otherwise you may expose controls that are not functional.

the hdac_hdmi_jack_init() function has no relationship with the HDMI Jack kctl. They are totally doing the different things. snd_jack_add_new_kctl() only depends on the "snd_soc_card_jack_new()".

Reply:
@libinyang what do you mean that they are doing different things, they use the same argument and both have a _jack_ in the function name. Can you please elaborate. This may be very clear to you but if you are not sharing the information we don't progress as a team. Thanks

Reply:
@plbossart hdac_hdmi_jack_init() main task is creating hdac_hdmi_pcm instance and calling snd_hdac_add_chmap_ctls(). It is creating a normal channel map ctls.

Reply:
A jack kctl is a  instance of struct snd_jack_kctl.

Reply:
so you are saying the function name hdac_hdmi_jack_init is misleading, should we rename it then?

Reply:
Yes, I think the name is not very good. However, it does do a little thing related jack init: pcm->jack = jack to assign the jack to hdac_hdmi_pcm.

Reply:
@libinyang your circular explanations leave me scratching my head. The function has nothing to do with jack, except that  it "assigns the jack to hdac_hdmi_pcm". 
I am going to approve and move on, can't keep doing this.